drivers/bridge/setup_device.go

```diff
@@ -25,7 +26,10 @@ func setupDevice(config *networkConfiguration, i *bridgeInterface) error {
 	// Only set the bridge's MAC address if the kernel version is > 3.3, as it
 	// was not supported before that.
 	kv, err := kernel.GetKernelVersion()
-	if err == nil && (kv.Kernel >= 3 && kv.Major >= 3) {
+	if err != nil {
+		logrus.Errorf("Failed to check kernel versions: %v. Will not assign a MAC address to the bridge interface", err)
+	}
+	if kv.Kernel > 3 || (kv.Kernel == 3 && kv.Major >= 3) {
```

Comment:
That's correct.  We should not even assume what the value of kv is in case of error and exclude this code.  Thanks.